cpp/tests/reductions/scan_tests.cpp

```diff
@@ -0,0 +1,476 @@
+
+    this->scan_test({v.begin(), v.end(), b.begin()}, 
+                    {exact.begin(), exact.end(), b.begin()},
+                    cudf::experimental::make_product_aggregation(), scan_type::INCLUSIVE);
+}
+
+TYPED_TEST(ScanTest, Sum)
+{
+    std::vector<TypeParam> v = {-120, 5, 6, 113, -111, 64, -63, 9, 34, -16};
+    std::vector<bool>      b = {   1, 0, 1,   1,    0,  0,   1, 1,  1,   1};
+    std::vector<TypeParam> exact(v.size());
+
+    std::partial_sum(
+        v.cbegin(),
+        v.cend(),
+        exact.begin(),
+        std::plus<TypeParam>{});
+
+    this->scan_test({v.begin(), v.end()}, 
+                    {exact.begin(), exact.end()},
+                    cudf::experimental::make_sum_aggregation(), scan_type::INCLUSIVE);
+
+    zip_scan(
+        v.cbegin(),
+        v.cend(),
+        b.cbegin(),
+        exact.begin(),
+        value_or<TypeParam>{0},
+        std::plus<TypeParam>{});
+
+    this->scan_test({v.begin(), v.end(), b.begin()}, 
+                    {exact.begin(), exact.end(), b.begin()},
+                    cudf::experimental::make_sum_aggregation(), scan_type::INCLUSIVE);
+}
+
+struct ScanStringTest : public cudf::test::BaseFixture {
+  void scan_test(cudf::test::strings_column_wrapper const& col_in,
+                 cudf::test::strings_column_wrapper const& expected_col_out,
+                 std::unique_ptr<aggregation> const &agg, scan_type inclusive) 
+  {
+    bool do_print = false;
+    if (do_print) {
+      std::cout << "input = {";  cudf::test::print(col_in);  std::cout<<"}\n";
+      std::cout << "expect = {";  cudf::test::print(expected_col_out);  std::cout<<"}\n";
+    }
+
+    const column_view input_view = col_in;
+    std::unique_ptr<cudf::column> col_out;
+
+    CUDF_EXPECT_NO_THROW(col_out = cudf::experimental::scan(input_view, agg, inclusive));
+    const column_view result_view = col_out->view();
+
+    cudf::test::expect_column_properties_equal(input_view, result_view);
+    cudf::test::expect_columns_equal(expected_col_out, result_view);
+
+    if (do_print) {
+      std::cout << "result = {"; cudf::test::print(result_view); std::cout<<"}\n";
+    }
+  }
+};
+
+TEST_F(ScanStringTest, Min)
+{
+    std::vector<std::string> v = {"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
+    std::vector<bool>        b = {    1,     0,       1,      1,      0,     0,       1,       0,      1};
+    std::vector<std::string> exact(v.size());
+
+    std::partial_sum(
+        v.cbegin(),
+        v.cend(),
+        exact.begin(),
+        [] (auto const& a, auto const& b) { return std::min(a, b); });
+
+    // string column without nulls
+    cudf::test::strings_column_wrapper col_nonulls(v.begin(), v.end());
+    cudf::test::strings_column_wrapper expected1(exact.begin(), exact.end());
+    this->scan_test(col_nonulls, expected1,
+                    cudf::experimental::make_min_aggregation(), scan_type::INCLUSIVE);
+
+    auto const STRING_MAX = std::string(1, char(127));
```

Comment:
UTF-8 max string character used in cudf
`__constant__ char max_string_sentinel[5]{"\xF7\xBF\xBF\xBF"};`

https://github.com/rapidsai/cudf/blob/branch-0.13/cpp/include/cudf/detail/utilities/device_operators.cuh#L85